server/base/src/main/java/org/apache/accumulo/server/util/CheckForMetadataProblems.java

```diff
@@ -125,7 +128,12 @@ private static void checkMetadataAndRootTableEntries(String tableNameToCheck, Se
         TreeSet<KeyExtent> tablets = tables.get(tableId);
         if (tablets == null) {
 
-          tables.forEach(CheckForMetadataProblems::checkTable);
+          for (Entry<TableId,TreeSet<KeyExtent>> e : tables.entrySet()) {
+            TableId key = e.getKey();
+            TreeSet<KeyExtent> value = e.getValue();
+            String tableName = Tables.getTableName(opts.getServerContext(), key);
+            checkTable(key, value, tableName);
+          }
```

Comment:
I think this line could have stayed the same and the name lookup could have occurred in the method. Also, need to handle the case that looking up the name fails, in case the table is deleted, but there is still metadata left in the metadata tablet. That wasn't an issue before because we only used the table ID before.

Reply:
>Also, need to handle the case that looking up the name fails, in case the table is deleted, but there is still metadata left in the metadata tablet

Is there any way that I would easily be able to recreate this error in testing?

Reply:
> Is there any way that I would easily be able to recreate this error in testing?

You could put the `Tables.getTableName` in a different method, and then do a partial mock or subclass this utility to provide a different response. However, it's not critical. Just wrapping `Tables.getTableName` with a try-catch block and setting tableName to null if there's an exception should be sufficient, as long as none of the rest of the code minds it being null.

Reply:
This should be addressed in my latest commit.